component/admin/models/wrappers.php

```diff
@@ -0,0 +1,121 @@
+/**
+ * @package     RedSHOP.Backend
+ * @subpackage  Model
+ *
+ * @copyright   Copyright (C) 2008 - 2019 redCOMPONENT.com. All rights reserved.
+ * @license     GNU General Public License version 2 or later; see LICENSE
+ */
+
+defined('_JEXEC') or die;
+
+/**
+ * Model Wrappers
+ *
+ * @package     RedSHOP.Backend
+ * @subpackage  Model
+ * @since       __DEPLOY_VERSION__
+ */
+class RedshopModelWrappers extends RedshopModelList
+{
+	/**
+	 * Construct class
+	 *
+	 * @since __DEPLOY_VERSION__
+	 */
+	public function __construct($config = array())
+	{
+		if (empty($config['filter_fields'])) {
+			$config['filter_fields'] = array(
+				'id',
+				'w.id',
+				'product_id',
+				'w.product_id',
+				'name',
+				'w.name',
+				'category_id',
+				'w.category_id',
+			);
+		}
+
+		parent::__construct($config);
+	}
+
+	/**
+	 * Method to auto-populate the model state.
+	 *
+	 * This method should only be called once per instantiation and is designed
+	 * to be called on the first call to the getState() method unless the model
+	 * configuration flag to ignore the request is set.
+	 *
+	 * Note. Calling getState in this method will result in recursion.
+	 *
+	 * @param string $ordering  An optional ordering field.
+	 * @param string $direction An optional direction (asc|desc).
+	 *
+	 * @return  void
+	 *
+	 * @since   __DEPLOY_VERSION__
+	 */
+	protected function populateState($ordering = 'w.id', $direction = 'asc')
+	{
+		$search = $this->getUserStateFromRequest($this->context . '.filter.search', 'filter_search');
+		$this->setState('filter.search', $search);
+
+		// List state information.
+		parent::populateState($ordering, $direction);
+	}
+
+	/**
+	 * Method to get a store id based on model configuration state.
+	 *
+	 * This is necessary because the model is used by the component and
+	 * different modules that might need different sets of data or different
+	 * ordering requirements.
+	 *
+	 * @param string $id A prefix for the store id.
+	 *
+	 * @return  string  A store id.
+	 *
+	 * @since   __DEPLOY_VERSION__
+	 */
+	protected function getStoreId($id = '')
+	{
+		$id .= ':' . $this->getState('filter.search');
+
+		return parent::getStoreId($id);
+	}
+
+	/**
+	 * Method to build an SQL query to load the list data.
+	 *
+	 * @return      string  An SQL query
+	 */
+	protected function getListQuery()
+	{
+		$db    = JFactory::getDbo();
+		$query = $db->getQuery(true)
+			->select('p.product_name,c.name,w.*')
+			->from($db->qn('#__redshop_wrapper', 'w'))
+			->leftJoin($db->qn('#__redshop_product', 'p') . ' ON p.product_id = w.product_id')
+			->leftJoin($db->qn('#__redshop_category', 'c') . ' ON c.id = w.category_id');
+
+		$search = $this->getState('filter.search');
+
+		if ($search) {
+			$query->where(
+				$db->qn('c.name') . ' LIKE ' . $db->q('%' . $search . '%') . ' OR ' .
+				$db->qn('w.name') . ' LIKE ' . $db->q('%' . $search . '%') . ' OR ' .
+				$db->qn('p.product_name') . ' LIKE ' . $db->q('%' . $search . '%')
+			);
+		}
```

Comment:
@hoang-tran-redweb please check query with old model wapper in method _buildQuery